src/resdk/resources/geneset.py

```diff
@@ -0,0 +1,118 @@
+"""Geneset resource."""
+import logging
+from collections.abc import Set
+
+from .data import Data
+
+
+class BaseGeneset(Set):
+    """Base class for Geneset resource.
+
+    :param genes: genes
+    :type genes: list of strings
+    """
+
+    def __init__(self, genes):
+        """Initialize genes list."""
+        self.genes = []
+        self.init(genes)
+
+    def init(self, genes):
+        """Initialize genes list.
+
+        :param genes: list
+        """
+        self.genes = []
+        if genes is not None:
+            for value in genes:
+                if value not in self.genes:
+                    self.genes.append(value)
+
+    def __iter__(self):
+        """Iterate genes."""
+        return iter(self.genes)
+
+    def __contains__(self, gene):
+        """Check if current genes list contains gene."""
+        return gene in self.genes
+
+    def __len__(self):
+        """Return number of genes."""
+        return len(self.genes)
+
+
+class Geneset(Data):
+    """Resolwe Geneset resource.
+
+    :param resolwe: Resolwe instance
+    :type resolwe: Resolwe object
+    :param model_data: Resource model data
+
+    """
+
+    WRITABLE_FIELDS = Data.WRITABLE_FIELDS + (
+        "genes",
+        "source",
+        "species",
+    )
+
+    def __init__(self, resolwe, **model_data):
+        """Initialize attributes."""
+        self.logger = logging.getLogger(__name__)
+
+        self._genes = None
+        self.source = None

```

Comment:
Aster some thinking i believe it would be best to also implement source and species as getter/setter methods. In case geneset is created from scratch, locally these method set the `self._species/_source` attributes. In case it is pulled from the server, the values for species/source are pulled from `Geneset` (`Data`) outputs: `self.output["source/species"]`

How do you differ between Geneset was/is pulled from the server vs. created locally? By checking the `self.id` attribute. Similar approach is used here:
https://github.com/genialis/resolwe-bio-py/blob/master/src/resdk/resources/base.py#L124

Reply:
Thanks for that idea. I implemented getter/setter methods for source and species.

With the last commit `gs = res.geneset.create(genes=[...], species=..., source=...)` should work. 
I updated some unit tests but it would probably be a good idea to add some functional tests as well.